private/model/api/shape.go

```diff
@@ -233,7 +245,7 @@ func goType(s *Shape, withPkgName bool) string {
 			return fmt.Sprintf("*%s.%s", pkg, s.ShapeName)
 		}
 		return "*" + s.ShapeName
-	case "map":
+	case "map", "jsonvalue":
 		return "map[string]" + s.ValueRef.GoType()
```

Comment:
Previous comment covers this where I think we can not treat `jsonvalue` as map at all ad more similar to a scaler or opaque type that is not transversed.